onnxscript/tests/function_libs/torch_lib/ops_test_data.py

```diff
@@ -836,14 +836,18 @@ def _where_input_wrangler(
         core_ops.aten_index_put_bool,
     ).skip(
         matcher=lambda sample: not (sample.args[0][0].dtype == torch.bool),
-        reason="this Aten overload only support tensor(bool) as args",
+        reason="this Aten overload only support tensor(bool) as indices",
     ),
     TorchLibOpInfo(
         "index_put",
         core_ops.aten_index_put,
     ).skip(
-        matcher=lambda sample: not (sample.args[0][0].dtype == torch.int64),
-        reason="this Aten overload only support tensor(int) as args",
+        matcher=lambda sample: not (
+            (sample.args[0][0].dtype == torch.int64)
+            # onnxruntime: MLFloat16 data type is not supported with ScatterND when reduction is 'add'
+            and (sample.args[1].dtype != torch.float16 or not sample.kwargs.get("accumulate", False))
+        ),
+        reason="this Aten overload only support tensor(int) as indices or float16 when reduction is 'add'",
```

Comment:
I tested this matcher with the original code and it shows no different in results (still 4 passes and 20 skipped). It seems that you found a corner case that is not in the current `op_DB_test`? If that's the case, we usually use https://github.com/microsoft/onnxscript/blob/main/onnxscript/tests/function_libs/torch_lib/extra_opinfo.py to create corner case to test the implementation.

Reply:
I did. I hope I did it right.